### PR TITLE
Add public share to blobber

### PR DESCRIPTION
**Pull request by lpoli**
This PR contains following changes:
1. Simply authticket generation to single function
2. Add public share to blobber. Previous public share was unrevokable.
3. Reduce conversions.

Linked to blobber changes https://github.com/0chain/blobber/pull/528

### CODE REVIEW

File: zboxcore/sdk/allocation.go [hunk 1]
```diff
@@ -1053,72 +1048,94 @@ func (a *Allocation) GetAuthTicket(
...
+	// 	}
+	// 	sEnc := base64.StdEncoding.EncodeToString(atBytes)
+	// 	return sEnc, nil
+	// }
+
+	// authTicket, err := shareReq.GetAuthTicket(refereeClientID)
+	// if err != nil {
+	// 	return "", err
+	// }
+	// return authTicket, nil
```
Comment (Kishan-Dhakan) [inline zboxcore/sdk/allocation.go]: Please remove commented code before merge (here + below)

**Review by Kishan-Dhakan — DISMISSED** [review]

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | master  |
| gosdk  | fix/public-share-issue  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | modify/download-test  |
| gosdk  | fix/public-share-issue  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | fix/public-share-issue  |
| gosdk  | fix/public-share-issue  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | modify/download-test  |
| gosdk  | fix/public-share-issue  |
| zboxcli  | fix/public-share-issue  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Review by dabasov — APPROVED** [review]
lgtm

**Review by cnlangzi — DISMISSED** [review]
@lpoli  we need get system-test passed first.

**Comment by service-0chain** [comment]
Manual system tests [success] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | modify/download-test  |
| gosdk  | fix/public-share-issue  |
| zboxcli  | fix/public-share-issue  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Pull request merged**